basebibliotek-fetch-cronjob/src/main/java/no/sikt/BasebibliotekFetchHandler.java

```diff
@@ -84,24 +80,39 @@ public BasebibliotekFetchHandler(S3Client s3Client, HttpClient httpClient, Envir
         this.basebibliotekAuthorization = createAuthorization();
     }
 
-    private String createAuthorization() {
-        String loginPassword = basebibliotekUsername + USERNAME_PASSWORD_DELIMITER + basebibliotekPassword;
-        return String.format(BASIC_AUTHORIZATION, Base64.getEncoder().encodeToString(loginPassword.getBytes()));
-    }
-
     @Override
-    public Set<String> handleRequest(ScheduledEvent scheduledEvent, Context context) {
+    public List<List<String>> handleRequest(ScheduledEvent scheduledEvent, Context context) {
         return attempt(() -> getBasebibliotekData(UriWrapper.fromUri(basebibliotekUri).getUri()))
                    .map(this::getBodyFromResponse)
                    .map(this::snipIncrementalBasebibliotekUrls)
                    .map(this::fetchBasebibliotekXmls)
                    .map(this::convertXmls)
                    .map(this::collectBibnrFromBaseBibliotek)
-                   .map(this::putObjectToS3)
+                   .map(this::convertToListOfListOfBibNr)
+                   .map(this::putObjectsToS3)
                    .orElseThrow(
                        fail -> logExpectionAndThrowRuntimeError(fail.getException(), fail.getException().getMessage()));
     }
 
+    private String createAuthorization() {
+        String loginPassword = basebibliotekUsername + USERNAME_PASSWORD_DELIMITER + basebibliotekPassword;
+        return String.format(BASIC_AUTHORIZATION, Base64.getEncoder().encodeToString(loginPassword.getBytes()));
+    }
+
+    private List<List<String>> convertToListOfListOfBibNr(Set<String> bibNr) {
+        List<List<String>> result = new ArrayList<>();
+        List<String> bibNrs = new ArrayList<>(bibNr);
+
+        var startIndex = 0;
+        while (startIndex < bibNrs.size()) {
+            var endIndex = Math.min(startIndex + 100, bibNrs.size());
```

Comment:
Kan kanskje være greit at vi lager en konstant for her (100)

Reply:
Fikset